magda-sleuther-format-layerer/src/format-engine/measures.ts

```diff
@@ -0,0 +1,196 @@
+import { Measure } from "@magda/typescript-common/src/format/Measure";
+import { Record } from "@magda/typescript-common/src/generated/registry/api";
+import { SelectedFormats } from "../../../magda-typescript-common/src/format/formats";
+import * as mimeTypes from "mime-types";
+import { Formats } from "@magda/typescript-common/src/format/formats";
+
+/*
+* Tries to determine the format by downloading the downloadURL, and deciphering the MIME type
+* TODO not unit tested
+*/
+export class DownloadMeasure extends Measure {
+    public relatedDistribution: Record;
+    getSelectedFormats(): SelectedFormats {
+        const { downloadURL } = this.relatedDistribution.aspects[
+            "dcat-distribution-strings"
+        ];
+        const rawMime: string = mimeTypes.lookup(downloadURL);
+
+        const processedMime: string = rawMime.split(rawMime)[1];
+
+        this.canDeduceFormat = true;
+        return {
+            selectedFormats: [
+                {
+                    format: (<any>Formats)[processedMime],
+                    correctConfidenceLevel: 100,
+                    distribution: this.relatedDistribution
+                }
+            ]
+        };
+    }
+}
+
+/*
+* Tries to determine the format by deciphering DCAT-DISTRIBUTION-STRING -> format
+* TODO Unit tests
+*/
+export class DcatFormatMeasure extends Measure {
+    public relatedDistribution: Record;
+
+    private applyTransform(format: string, transformFormat: any) {
+        const newFormat = transformFormat(format);
+        return newFormat;
+    }
+
+    private empty(str: string) {
+        return "";
+    }
+
+    private lowerCase(str: string) {
+        return str.toLowerCase();
+    }
+
+    private removePeriod(str: string) {
+       return str.split('.').join('');
+    }
+
+    private replaceCommaThenSpace(str: string) {
+        return str.split(', ').join(' ');
+    }
+
+    private replaceSpaceThenComma(str: string) {
+        return str.split(' ,').join(' ');
+    }
+
+    private replaceComma(str: string) {
+        return str.split(',').join(' ');
+    }
+
+    private identity(str: string) {
+        return str;
+    }
+
+    getSelectedFormats(): SelectedFormats {
+        const { format } = this.relatedDistribution.aspects[
+            "dcat-distribution-strings"
+        ];
+
+        if(format === null || format === "") {
+            this.canDeduceFormat = false;
+            return null;
+        }
+
+        let processedFormat: string;
+
+        const transforms = [
+            this.identity,
+            (str: string) => this.lowerCase(this.removePeriod(this.replaceComma(this.replaceSpaceThenComma(this.replaceCommaThenSpace(str)))))
```

Comment:
This looks like it's equivalent to `str.toLowerCase().replace(/\s*[.,]\s*/g, ' ')` ?